src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -565,7 +565,6 @@ bool reportNullLiteralAssignmentIfNecessary()
         private void ReportAssignmentWarnings(BoundExpression value, TypeSymbolWithAnnotations targetType, TypeSymbolWithAnnotations valueType, bool useLegacyWarnings)
         {
             Debug.Assert(value != null);
-            Debug.Assert(!IsConditionalState);
```

Comment:
If we have a split state, does that mean we'll report duplicate warnings?

Reply:
I would think we'd only get duplicate warnings if we visit the same expression multiple times.

---
In reply to: [211798748](https://github.com/dotnet/roslyn/pull/29429#discussion_r211798748) [](ancestors = 211798748)